pkg/tsdb/postgres/postgres.go

```diff
@@ -47,20 +47,47 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	timescaledb := datasource.JsonData.Get("timescaledb").MustBool(false)
 
 	endpoint, err := sqleng.NewSqlQueryEndpoint(&config, &queryResultTransformer, newPostgresMacroEngine(timescaledb), logger)
-	if err == nil {
-		logger.Debug("Successfully connected to Postgres")
-	} else {
+	if err != nil {
 		logger.Debug("Failed connecting to Postgres", "err", err)
+		return nil, err
 	}
+
+	logger.Debug("Successfully connected to Postgres")
 	return endpoint, err
 }
 
 func generateConnectionString(datasource *models.DataSource, logger log.Logger) (string, error) {
+	logger.Debug("Trying to generate Postgres connection string")
+
 	sslMode := strings.TrimSpace(strings.ToLower(datasource.JsonData.Get("sslmode").MustString("verify-full")))
 	isSSLDisabled := sslMode == "disable"
 
-	// Always pass SSL mode
-	sslOpts := fmt.Sprintf("sslmode=%s", url.QueryEscape(sslMode))
+	var host string
+	var port int
+	if strings.HasPrefix(datasource.Url, "/") {
+		host = datasource.Url
+		logger.Debug("Generating connection string with Unix socket specifier", "socket", host)
+	} else {
+		sp := strings.SplitN(datasource.Url, ":", 2)
```

Comment:
still think this can be simplified by moving the url check to the bottom, if unix socket return connectionstring without url encode, if not return `&url.URL{...}.String()` as before since it includes url encode.

Reply:
@marefr I don't understand the point of using a connection URL here at all, and respectfully disagree that we should use it if the host is not a socket. For one, there's no need to use the URL format (it makes things trickier, and the non-URL connection string is [more standard](https://pkg.go.dev/github.com/lib/pq?tab=doc)), two, using either the URL format or the normal connection string format depending on whether a socket is involved is more complicated logic, and makes for more difficult debugging.